File: Data Structures & Algorithms/maximum-product-subarray/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxProduct(nums) {
        let n= nums.length;
        if(n===1) return nums[0];
        let res =0;
        for(let i=0;i<n;i++){
        let curr=1;
        let j=i;
        while(j<n){
          curr*=nums[j];
          j++;
          res=Math.max(res,curr);
        }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count={}
        l=0
        res=0
        mx=0
        for r in range(len(s)) :
            if s[r] not in count :
                count[s[r]]=1
            else :
                count[s[r]]+=1
            mx=max(mx,count[s[r]])
            while r - l + 1 - mx > k :
                count[s[l]]-=1
                l+=1
            
            res=max(res,r-l+1)     
        return res